### ISSUE TITLE
Option for default directory missing

### ISSUE THREAD

**Issue by AmarjeetBanwait** [desktop/desktop#2889]
In new Github Desktop there is no option for changing default directory for cloning repos.

**Comment by LucDorpmans** [comment]
@Shiftkey: I don't observe the behaviour you describe. Just added 5 repo's to a local drive with a different path than %UserProfile%\Documents\GitHub'  and had to specify the location every time. I would like to specify the default location.

**Comment by iAmWillShepherd** [comment]
@LucDorpmans can you open a new issue for this?

**Comment by KartikGupta1997** [comment]
My local disk C was almost full. The best i could do was to change the default directory of Documents folder

**Comment by frumbert** [comment]
Mine always tries to clone into
`/Users/tim/Documents/GitHub`

whereas all my gits are stored in 
`/Users/tim/GitHub`

changing it and performing a new clone still makes the default
`/Users/tim/Documents/GitHub`

I don't observer the behaviour described in #issuecomment-332415957

Why cant it be a settable option? Github desktop 1.5.0, macos

**Comment by epodox** [comment]
I can confirm this is still an issue many months later. Has there been any update on this functionality?

**Comment by patch-af** [comment]
Has this been fixed?

**Comment by vogtc3** [comment]
I’m not sure if this has been fixed but I do know that using the command line instead of desktop results in it being saved in the user folder instead of the iCloud documents folder on macOS

**Comment by edwardbattistini** [comment]
same here...

**Comment by some1new** [comment]
Why is it so difficult to give users an option to set default folder?

**Comment by habibieamrullah** [comment]
yeah I also need an option in GitHub Desktop to change default folder / path of my coding project.

**Comment by habibieamrullah** [comment]
I found a workaround for this issue, which is to delete or move the current project folder.. then next time we open GitHub Desktop it will ask to locate the missing directory.. so we can point to new directory that we wish to use.

**Comment by Tsubakura** [comment]
Same here. Consider the scenario where people might have 2 drives, it makes absolutely no sense to force people into using the smaller primary drive where Windows is installed, instead of a larger secondary drive.

It has been 3 years since this issue was made and I cant fathom as to why nobody thought of adding an option to set the default folder.

**Comment by az6667** [comment]
Hmm. how badly does this annoy everyone else? I'll do it if it'll help and if I remember it over the weekend.
"last-clone-location" isn't all that useful - a default would be nice.

**Comment by harriswe** [comment]
Yikes I guess GitHub really hit the sticks when Microsoft took over.

**Comment by ocervell** [comment]
+1, this is mandatory to have ! We don't want to browse to our Git location everytime we clone something !!!

**Comment by johnlewisdt** [comment]
Here we are, 4 years later nearly, with no action on this point. Small mercy is that you can create a new repo called '**now-theyre-microsoft-good-luck-with-support**' and set it to users/me/github, then it remembers it next time.

**Comment by michael-crawford** [comment]
I have also been mystified why there's no preferences option to set this. It's not only really useful, it seems like a realoversight to not support such an obvious option, requested by many users. Is there some reason something that's likely very simple to implement is being ignored?

Ideally any preference should support variables which can be substituted from the GitHub URL.  For example. I keep my repos outside of ~/Documents as I don't want my per-workstation changes to be replicated elsewhere via iCloud or OneDrive, as I'm using GitHub to coordinate.

So, my MacOS/Windows convention is:
~/Workspaces/${organization}/${repo}

However on Linux I use:
~/src/${organization}/${repo}
(with a symlink so ~/Workspaces also works, just more conventional to use src on linux)

Note I don't want to use .../GitHub/... as I used to also use SourceTree, some clients use GitLab, so want something generic. All  have an organization parent.

Because sometimes I work with forks that exist under another organization with the same name, or I have both GitHub and some GitHub Enterprise sources, at a minimum, I'd like to have a way to grab the organization from the clone url and use that to modify the base.

Although I don't use it, it would be good to have a way to have per-source roots, in case there was a need to clone GitHub public website org/repo to a different base location than one or more internal GitHub Enterprise private org/repo code, as some of that code could be restricted by contract or very sensitive in nature, and need to go on a different drive or filesystem.

**Comment by vigji** [comment]
This is just unbelievable. Such a simple addition, so many people asking it...would be amazing to have for me as well

**Comment by bmcmcm** [comment]
There should be an option to specify a default directory. Having the default location be in the currently logged on user's documents is a problem. I work in the context of a regular user account but frequently need to run code on an administrator console. Having the default location being in a user context results in having to navigate to a different user context in the file system to run the code during testing.

I cannot fathom why a default location cannot be specified.

**Comment by chenhajaj** [comment]
+1

**Comment by bhavisheythapar** [comment]
Would really like this option. +1

**Comment by Charles-Svg** [comment]
Can the default clone directory be modified with the Git comand line ? 
That would solve my prob

**Comment by az6667** [comment]
okay, adding this feature/fix sometime this week, hopefully they take the pull request

**Comment by ecocarlisle** [comment]
Is there a workaround for this?

**Comment by steveward** [comment]
Just to reiterate what was mentioned in the initial reply to this -- GitHub Desktop remembers the directory you previously cloned/created a repository to, so if you clone to `~/Code` that directory will be selected by default the next time you clone/create a repository.

**Comment by ans-human** [comment]
Patiently waiting for the solution to this.

**Comment by pearson** [comment]
While searching for an answer to the default directory question, the Google led me to this issue. What has been stated here seems to be correct (at least for me) and GitHub Desktop is remembering the previous directory I cloned a repository too. TBH, I wanted to move the GitHub directory, I looked at the settings in GitHub Desktop and didn't find a default directory option, and Googled right away -- it did not occur to me that the software would remember the previous directory.

That said, @shiftkey, given how many people are asking for it and given that the current functionality is a hidden feature, perhaps adding a default directory option to the settings would not be a bad idea? A UI that's obvious to the masses is a good thing. :smiley:

As well, perhaps there's some case where GitHub Desktop is not remembering the previous directory? There are a lot of "thumbs up" on comments stating that the software reverts to `%UserProfile%\Documents\GitHub`.

**Comment by mcquiggd** [comment]
GitHub badly needs to hire some UX and Product Owner talent; basics such as being able to set a default location for creating a new repo should never require users to request it repeatedly, for years. 

It's the same with being able to list repos by name - it took four years and is only available for organisations, not yet for personal repos.

Meanwhile, we have fluff like animated profile READMEs. being prioritised... 

It's baffling, frankly.

**Comment by LindsayMoir** [comment]
Still no option for this. Not very pythonic, that is for sure (sigh).

**Comment by nateGeorge** [comment]
It's possible, you need to change the directory when cloning a repo. Then all future repos will use that by default: https://stackoverflow.com/a/68474257/4549682

**Comment by nikhilagrawaldotnet** [comment]
I am on v2.9.0 (x64) and it worked as described by @shiftkey 

1. Tried to clone a repo.
2. GH gave its default location. 
3. Cloned repo in my location.
4. Cloned a new repo and it gave my location.

This worked without closing GH Desktop & also after closing and reopening GH Desktop.

Now, we have couple of solutions to close this -
1. Give default location option (destination will be decided).
![image](https://user-images.githubusercontent.com/26463468/127097266-64294103-e58a-48c9-a2b2-310c721eed5f.png)

2. On the clone option, give additional info of this hidden feature.
![image](https://user-images.githubusercontent.com/26463468/127097176-0ac729e5-2c23-4ca8-983d-b3182633dbf2.png)

**Comment by bhavisheythapar** [comment]
The 2nd option would work.

**Comment by FlowerForWar** [comment]
+1

**Comment by SPOOKEXE** [comment]
Yep definitely would like the second option, still non-existent to this day ;(

**Comment by Incredible-IT** [comment]
Use the OS to create a symbolic link. 

rem Close GitHub Desktop first.
rmdir "%USERPROFILE%\Documents\GitHub" /S /Q
mklink /J "%USERPROFILE%\Documents\GitHub" "D:\Code"
